### PR TITLE
adding torrent search cog

### PR DESCRIPTION
**Pull request by rssnyder**
Add ability to search for torrents on TPB.

Usage: `!torrent last week tonight 420p`

Posts magnet link, or link to TPB if no results returned.

### CODE REVIEW

File: cogs/torrents.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    failure = discord.Embed(
+        title='Unable to find torrent',
+        url=f'https://www.thepiratebay.org/search/{search}',
+        description='No Magnet Found'
+    )
+    failure.set_footer(text='Better luck next time!')
+
+    try:
+        response = requests.get(f'https://apibay.org/q.php?q={search}')
+    except Exception:
```
Comment (Joelpls) [inline cogs/torrents.py]: This exception should be more explicit

File: cogs/torrents.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    # Top torrent magnet link
+    desc = f"Magnet: magnet:?xt=urn:btih:{target.get('info_hash')}&dn={target.get('name')}\n\n"
+    desc += f"Seeders: {target.get('seeders')}\nLeachers: {target.get('leechers')}\n"
+    desc += f"Uploader: {target.get('username')}\nSize:  {target.get('size')}"
+
+    success = discord.Embed(
+        title=target.get('name'),
+        description=desc
+    )
+    success.set_footer(text='I am not responsible for anything downloaded via this cog')
```
Comment (Joelpls) [inline cogs/torrents.py]: change cog to command.
Does it even need this as a footer tho?

File: cogs/torrents.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        config = json.load(f)
+    return config.get(token)
+
+
+def setup(client):
+    client.add_cog(Torrents(client))
+
+
+def get_magnet(search: str) -> discord.Embed:
+    """Get a magnet link for a torrent"""
```
Comment (Joelpls) [inline cogs/torrents.py]: Add Usage to the help comments. See https://github.com/Joelpls/discord-bot/blob/1488dca517c77d15962b16358f473c8d53e70f83/cogs/memes.py#L37

File: cogs/torrents.py (hunk 2 above)
Comment (rssnyder) [reply cogs/torrents.py]: Probably not. Removed.

File: cogs/torrents.py (hunk 1 above)
Comment (rssnyder) [reply cogs/torrents.py]: Right, are we just logging via stdout?

Comment (Joelpls) [reply cogs/torrents.py]: I don't do much logging. Don't tell anyone.

File: cogs/torrents.py [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    client.add_cog(Torrents(client))
+
+
+def get_magnet(search: str) -> discord.Embed:
+    """
+    Get a magnet link for a torrent.
+    Uses the backend api for 'the pirate bay' to find torrent hash and name.
+    Create the magnet link based on the information returned.
+    Present to the user with helpful information.
+    """
```
Comment (Joelpls) [inline cogs/torrents.py]: I think something better is:
Get a magnet link for a torrent.
Usage: !torrent <search>

Something like that. It doesn't need all the technical details. This is what shows up when the user types !help torrent

Comment (rssnyder) [reply cogs/torrents.py]: Ahhhhh ok makes sense.

**Comment by rssnyder** [comment]
Whats the hold up, I want to start browsing!

**Pull request closed without merging**